cwms-data-api/src/main/java/cwms/cda/api/location/kind/GateChangeGetAllController.java

```diff
@@ -0,0 +1,117 @@
+            },
+            queryParams = {
+                    @OpenApiParam(name = BEGIN, required = true, description = "The start of the time window"),
+                    @OpenApiParam(name = END, required = true, description = "The end of the time window."),
+                    @OpenApiParam(name = START_TIME_INCLUSIVE, type = Boolean.class, description = "A flag "
+                            + "specifying whether any data at the start time should be retrieved ('True') "
+                            + "or only data <b><em>after</em></b> the start time ('False').  "
+                            + "Default value is True"),
+                    @OpenApiParam(name = END_TIME_INCLUSIVE, type = Boolean.class, description = "A flag "
+                            + "('True'/'False') specifying whether any data at the end time should be "
+                            + "retrieved ('True') or only data <b><em>before</em></b> the end time ('False'). "
+                            + "Default value is False"),
+                    @OpenApiParam(name = UNIT_SYSTEM, type = UnitSystem.class, description = "Unit System desired in response. "
+                            + "Can be SI (International Scientific) or EN (Imperial.) If unspecified, "
+                            + "defaults to EN."),
+                    @OpenApiParam(name = PAGE_SIZE, type = Integer.class,
+                            description = "the maximum number of gate changes to retrieve, regardless of time window. " +
+                                    "A positive integer is interpreted as the maximum number of changes from the " +
+                                    "beginning of the time window. A negative integer is interpreted as the maximum number " +
+                                    "from the end of the time window. " + "Default " + DEFAULT_PAGE_SIZE + "." +
+                                    "A page cursor will not be returned by this DTO. Instead, the next page can be determined " +
+                                    "by querying the next set of changes using the last returned change date and using " +
+                                    "start-time-inclusive=false")
+            },
+            responses = {
+                    @OpenApiResponse(status = STATUS_200, content = {
+                            @OpenApiContent(from = GateChange.class, isArray = true, type = Formats.JSONV1)
+                    })
+            },
+            description = "Returns matching CWMS gate change data for a Reservoir Project.",
+            tags = {OutletController.TAG}
+    )
+    @Override
+    public void handle(@NotNull Context context) {
+        String office = context.pathParam(OFFICE);
+        String id = context.pathParam(PROJECT_ID);
+        CwmsId projectId = new CwmsId.Builder().withName(id).withOfficeId(office).build();
+        Timestamp startTime = Timestamp.from(requiredZdt(context, BEGIN).toInstant());
```

Comment:
Instead of using `Timestamp` use `requiredInstant` and map to `Timestamp` at the jOOQ level.

Reply:
Oh, didn't know that was a thing, thanks.